presto-main/src/test/java/com/facebook/presto/server/remotetask/TestHttpRemoteTask.java

```diff
@@ -129,8 +144,18 @@ public void testRemoteTaskMismatch()
 
         httpRemoteTaskFactory.stop();
         assertTrue(remoteTask.getTaskStatus().getState().isDone(), format("TaskStatus is not in a done state: %s", remoteTask.getTaskStatus()));
-        assertEquals(getOnlyElement(remoteTask.getTaskStatus().getFailures()).getErrorCode(), REMOTE_TASK_MISMATCH.toErrorCode());
         assertTrue(remoteTask.getTaskInfo().getTaskStatus().getState().isDone(), format("TaskInfo is not in a done state: %s", remoteTask.getTaskInfo()));
+
+        switch (testCase) {
+            case TASK_MISMATCH:
+                assertEquals(getOnlyElement(remoteTask.getTaskStatus().getFailures()).getErrorCode(), REMOTE_TASK_MISMATCH.toErrorCode());
```

Comment:
maybe assign `getOnlyElement(remoteTask.getTaskStatus().getFailures()).getErrorCode()` to a variable